File: durmitor/src/main/java/ql/ast/statement/ComputedQuestion.java
package ql.ast.statement;

import ql.ast.expression.Expression;
import ql.ast.expression.Identifier;
import ql.ast.type.Type;
import ql.visitors.interfaces.StatementVisitor;

public class ComputedQuestion extends Question {
    
    private Expression expr;

    public ComputedQuestion(String label, Identifier id, Type type, Expression expr) {
        super(label, id, type);
        this.expr = expr;
    }

    @Override
    public String toString() {
        return "\"" + label.toString() + "\" " + id.toString() + ": " + type.toString() + "( " + expr.toString() + " )";
    }

    @Override
    public void accept(StatementVisitor visitor) {
        visitor.visit(this);
    }
    
    public Expression getExpression() {
        return expr;
    }
}

File: Erik_Vincent/Assignment1/Assignment1/Model/QL/AST/ComputedQuestion.cs
﻿using Assignment1.Model.QL.AST.Expression;

namespace Assignment1.Model.QL.AST
{
    public class ComputedQuestion : Question
    {
        public IExpression Computation { get; }

        public ComputedQuestion(int lineNumber, string id, string label, Type type, IExpression computation) : base(lineNumber, id, label, type)
        {
            Computation = computation;
        }

        public override void Accept(IQLASTVisitor visitor) => visitor.Visit(this);
    }
}
